ci-operator/step-registry/cucushift/installer/check/ibmcloud/resourcegroup/cucushift-installer-check-ibmcloud-resourcegroup-commands.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+#OCP-60816 - [IPI-on-IBMCloud] install cluster under BYON with a different resource group
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC() {
+    local preResourceFile resource_group curResourceFile cmpRst
+    resource_group=$1
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "diff ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(diff ${curResourceFile} ${preResourceFile})
+    if [ -n "$cmpRst" ]; then
+        echo "ERROR: resources differ: ${cmpRst}"
+        return 1
+    else
+        echo "checkResourcesinVPC PASS"  
+        return 0
+    fi
+}
+
+cluster_rg_file="${SHARED_DIR}/ibmcloud_cluster_resource_group"
+provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+
+if [ ! -f "${cluster_rg_file}" ] && [ ! -f "${provisioned_rg_file}" ]; then
```

Comment:
Just like what I explained, we already introduced this dedicated step to run dedicated resource group, if the install workflow never use any provisioned resource group, then the step should never be inserted into the install workflow.